test/server.ts

```diff
@@ -237,6 +244,135 @@ test('store server started in the background should use store location wanted by
   t.notOk(await pathExists(serverJsonPath), 'server.json removed')
 })
 
+interface ChildProcess extends EventEmitter {
+  pid: number;
+  stderr: Readable;
+  stdout: Readable;
+}
+
+type ServerProcessList = Array<[ChildProcess, Deferred<void>]>
+
+async function testParallelServerStart (options: {
+  onTimeout?: (serverProcessList: ServerProcessList) => Promise<void>,
+  timeoutMillis?: number,
+  onProcessStarted?: (serverProcess: ChildProcess, serverProcessDeferred: Deferred<void>) => void,
+  onProcessClosed?: (serverProcess: ChildProcess) => void,
+  }) {
+  const pnpmBin = 'pnpm'
+
+  // List of pairs of ChildProcess, Deferred
+  const serverProcessList: ServerProcessList = []
+
+  let timeoutObj: number|NodeJS.Timer|null = setTimeout(() => {
```

Comment:
why don't you start the timer at the end of the function? Also, you could just use `await delay(...`